engine/robot.py

```diff
@@ -366,49 +377,48 @@ def track_obstacle(self):
         front_ultrasonic = None
         counter = 0  # added for testing
         while True:
-            if self.robot_state.is_sim:
-                ultrasonic_value_file = open(ROOT_DIR + '/tests/functionality_tests/csv/ultrasonic_values.csv',
-                                             "r")  # added for testing
-                content = ultrasonic_value_file.readlines()
-                ultrasonic_value_file.close()
-                try:
-                    line = content[counter]
-                    counter += 1
-                    curr_ultrasonic_value = float(
-                        (''.join(line.rstrip('\n')).strip('()').split(', '))[0])
-                except IndexError:
-                    print("no more sensor data")
-                    break
-            else:
-                from electrical.ultrasonic_sensor import Ultrasonic
-                front_ultrasonic = Ultrasonic(0)
-                curr_ultrasonic_value = front_ultrasonic.distance()
-                if curr_ultrasonic_value < self.robot_state.front_sensor_offset:
-                    self.set_phase(Phase.FAULT)
-                    return None
-            if self.robot_state.is_roomba_traversal:  # roomba mode
-                self.robot_state.is_roomba_obstacle = True
-            elif (self.robot_state.phase == Phase.TRAVERSE) or (self.robot_state.phase == Phase.RETURN) or (self.robot_state.phase == Phase.DOCKING) or (
-                    self.robot_state.phase == Phase.AVOID_OBSTACLE):
-                if curr_ultrasonic_value < self.robot_state.detect_obstacle_range:
-                    # Note: didn't check whether we can reach goal before contacting obstacle because obstacle
-                    # detection does not detect angle, so obstacle could be calculated to be falsely farther away than
-                    # the goal. Not optimal because in cases, robot will execute boundary following when it can reach
-                    # goal
-                    self.set_phase(Phase.AVOID_OBSTACLE)
-                    if self.robot_state.is_sim:
-                        with open(ROOT_DIR + '/tests/functionality_tests/csv/avoid_obstacle_result.csv', 'a') as fd:
-                            fd.write("Avoid" + '\n')
+            if self.robot_state.enable_obstacle_avoidance:
```

Comment:
Lmk what you think about this but since we are just nesting all the below code under this if statement i think it would be cleaner if separated out the opposing condition and handled it earlier to avoid nesting if statements. Thinking about the idea in 2) of the docking algorithm [here](https://github.com/cornellnexus/src/pull/120#pullrequestreview-1349568065) -- though the decision to guard here isn't quite as clean bc we still need the while True loop to keep looping (normally would just negate and return from the function early)

What do you think about this?

```
if not self.robot_state.enable_obstacle_avoidance:
    pass
if self.robot_state.is_sim:... [prev code]
```
Not sure which is better here bc i think in general reducing nested ifs when possible is cleaner but idk if the pass is weird.

Reply:
uhhh, seems a bit weird for me but I'm not opposed

Reply:
also, I think it should be continue, not pass

Reply:
whatever u think is best either fine!